File: solutions/kotlin/01-dr6/code/src/main/kotlin/Main.kt
import java.nio.ByteBuffer
import kotlin.system.exitProcess

fun main(args: Array<String>) {
    if (args.size != 2) {
        println("Usage: ./your_program.sh <database path> <command>")
        exitProcess(1)
    }

    val databaseFilePath = args[0]
    val command = args[1]

    if (command == ".dbinfo") {
        val databaseFile = java.io.File(databaseFilePath).inputStream()

        databaseFile.skip(16) // Skip the first 16 bytes of the header
        val pageSizeBytes = ByteArray(2) // The following 2 bytes are the page size
        databaseFile.read(pageSizeBytes)
        val pageSize = ByteBuffer.wrap(pageSizeBytes).short.toUShort()
        println("database page size: $pageSize")

        databaseFile.close()
    } else {
        println("Invalid command: $command")
    }
}

File: solutions/elixir/01-dr6/code/lib/main.ex
defmodule CLI do
  def main(args) do
    case args do
      [database_file_path, command] ->
        case command do
          ".dbinfo" ->
            file = File.open!(database_file_path, [:read, :binary])

            :file.position(file, 16) # Skip the first 16 bytes of the header
            <<page_size::16>> = IO.binread(file, 2)
            IO.puts("database page size: #{page_size}")
        end
      _ ->
        IO.puts("Usage: your_program <database_file> <command>")
    end
  end
end
